src/core.c

```diff
@@ -615,7 +616,15 @@ int connect_proxy_chain(int sock, ip_type target_ip,
 					/* We've reached the end of the list, go to the start */
  					offset = 0;
 					looped++;
-					continue;
+					if (looped > rr_loop_max) {
+						goto error_more;
+					} else {
+						PDEBUG("rr_type all proxies down, release all\n");
+						release_all(pd, proxy_count);
```

Comment:
why is release_all() necessary ?

Reply:
`select_proxy()` just finds an a proxy in `PLAY_STATE`,  and then tries to connect via this proxy. `release_all()` resets all proxies' state to `PLAY_STATE`
The changes have two purposes:
1. If all proxies failed, wait and retry, so we call `release_all()` before another select loop.
2. Fast fail, if all proxies failed, we don't wait too long. So I choose 14 loops sums up with 1.05 second
The original code is a bit confusing
1. In this if condition waits for an available proxy https://github.com/rofl0r/proxychains-ng/blob/master/src/core.c#L614-L619
2. While in this else block, https://github.com/rofl0r/proxychains-ng/blob/master/src/core.c#L619-L624  if all proxies have failed to connect in one loop, we will always run `goto error`

I pushed another commit to have better error handling

Reply:
where does the seemingly arbitrary number 14 come from ? is it some "lucky number" in china ?

Reply:
The number 14  has no special meaning. As I said first loop wait 10ms, then each loop increase 10ms, 14 loops sums up with 1.05s. When all proxies are not available, what we can do is either retry or raise an error. Do you have some better suggestions?

Reply:
well i thought a config variable would maybe be more suitable. but then it seems you're the only one using round robin mode anyway...